### PR TITLE
Connect Kube gateway part 3: daemon create kube gateway

### PR DESCRIPTION
**Pull request by greedy52**
Related issues
- Part 1: https://github.com/gravitational/teleport/pull/28312
- Part 2: https://github.com/gravitational/teleport/pull/28920

Changes:

- Daemon/Clusters to create kube gateway
- Complete kube reissue flow
- Integration test for kube gateway

This should complete all kube gateway changes on the `golong` side.

### CODE REVIEW

File: integration/proxy/teleterm_test.go [hunk 1]
```diff
@@ -26,67 +27,97 @@ import (
...
+
+	tc, err := inst.NewClient(helpers.ClientConfig{
+		Login:   username,
+		Cluster: inst.Secrets.SiteName,
 		ALBAddr: albAddr,
-	}, *creds)
+	})
 	require.NoError(t, err)
 	// Save the profile yaml file to disk as NewClientWithCreds doesn't do that by itself.
 	tc.SaveProfile(false /* makeCurrent */)
```
Comment (ravicious) [inline integration/proxy/teleterm_test.go:118]:
```suggestion
	err = tc.SaveProfile(false /* makeCurrent */)
	require.NoError(t, err)
```

I forgot about checking `err` here when I was writing this test.

File: integration/proxy/teleterm_test.go [hunk 2]
```diff
@@ -232,3 +233,129 @@ func (c *mockTSHDEventsService) SendNotification(context.Context, *api.SendNotif
 	c.callCounts["SendNotification"]++
 	return &api.SendNotificationResponse{}, nil
 }
+
+// TestTeletermKubeGateway tests making kube API calls against Teleterm kube
+// gateway and reissuing certs.
+//
+// Note that this test does NOT reuse existing kube test setups as IP Pinning
+// is enabled in those tests. Use certs with pinned IPs are injected during
```
Comment (ravicious) [inline integration/proxy/teleterm_test.go:241]: I think it was supposed to be "user certs" not "use certs", correct?

```suggestion
// is enabled in those tests. User certs with pinned IPs are injected during
```

File: lib/teleterm/daemon/daemon_test.go [hunk 3]
```diff
@@ -98,6 +101,7 @@ func TestGatewayCRUD(t *testing.T) {
 	tests := []struct {
 		name                 string
 		gatewayNamesToCreate []string
+		gatewayTargetURI     func(name string) uri.ResourceURI
```
Comment (ravicious) [inline lib/teleterm/daemon/daemon_test.go:104]:
```suggestion
		appendGatewayTargetToURI     func(name string) uri.ResourceURI
```

**Review by ravicious — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]
LGTM.

**Review by gzdunek — APPROVED** [review]

**Pull request merged**